utils/job_db.py

```diff
@@ -23,21 +33,24 @@ def start_db(self):
 
     def generate_unique_object_identifier(self):
         """
-        Create the unique object identifier.
+        Create the unique object identifier suffix.
 
-        This is NOT the whole object id, just the last part to ensure every object
+        This is not the whole object id, just the last part to ensure every object
         is unique and the last characters are digits.
         :return:
         """
         try:
-            return self.db.objects.find_one_and_update(
-                {'next_object_id': {'$exists': True}},
-                {'$inc': {'next_object_id': 1}},
+            # The document holding the ids is initially set up via start_db()
+            return  self.db.objects.find_one_and_update(
+                { '_id': 'object_ids'},
+                {'$inc': {'next': 1}},
+                projection={'_id': False, 'next': True},
                 return_document=ReturnDocument.AFTER
-            )['next_object_id']
+            )['next']
         except (KeyError, TypeError):
-            raise RuntimeError("The database doesn't seem to be initialized"
-                               "properly")
+            raise RuntimeError(
+                "Could not get object id. Is the job-db up and initialized?"
+            )
```

Comment:
Nur zum Verständnis: War der Wechsel von 'next_object_id' zu '_id' notwendig, oder einfach weil es klarer wird dass da dann nur ein spezieller Datensatz gesucht wird?

Reply:
Der Wechsel ist notwendig, damit es wirklich nur noch ein Dokument in der MongoDB geben kann, in dem die ids gepflegt werden. In der Test-MongoDB sind irgendwie vier Datensätze entstanden mit unterschiedlichen '_id'-Feldern, die alle 'next_object_id' haben. Das war so eigentlich nicht beabsichtigt und diese Änderung schließt aus, dass das noch einmal passieren kann.